Realm/RLMQueryUtil.mm

```diff
@@ -972,12 +1040,105 @@ void convert_null(T&& value, Fn&& fn) {
+                                     column.resolve<W<Mixed>>(),
+                                     value);
+            });
+    }
+}
+
+#pragma mark Mixed Constraints
+
+template<typename C, typename T>
+void QueryBuilder::add_mixed_constraint(NSPredicateOperatorType operatorType,
+                                        NSComparisonPredicateOptions predicateOptions,
+                                        Columns<C>&& column,
+                                        T value)
+{
+    if (auto i = RLMDynamicCast<NSNumber>(value)) {
+        if ([value objCType][0] == 'c') {
+            add_bool_constraint(RLMPropertyTypeAny, operatorType, column, value_of_type<Bool>(value));
+        }
+        else if (numberIsInteger(i)) {
+            add_numeric_constraint(RLMPropertyTypeAny, operatorType, column, value_of_type<Int>(value));
+        }
+        else if ([value objCType][0] == *@encode(float)) {
+            add_numeric_constraint(RLMPropertyTypeAny, operatorType, column, value_of_type<Float>(value));
+        }
+        else if ([value objCType][0] == *@encode(double)) {
+            add_numeric_constraint(RLMPropertyTypeAny, operatorType, column, value_of_type<Double>(value));
+        }
+    }
+    else {
+        add_mixed_constraint(operatorType,
+                             predicateOptions,
+                             std::move(column),
+                             value_of_type<Mixed>(value));
+    }
+}
+
+template<typename C>
+void QueryBuilder::add_mixed_constraint(NSPredicateOperatorType operatorType,
+                                        NSComparisonPredicateOptions predicateOptions,
+                                        Columns<C>&& column,
+                                        const ColumnReference& value)
+{
+    add_bool_constraint(RLMPropertyTypeObject, operatorType, column, value.resolve<Mixed>());
+}
+
+template<typename C>
+void QueryBuilder::add_mixed_constraint(NSPredicateOperatorType operatorType,
+                                        NSComparisonPredicateOptions predicateOptions,
+                                        Columns<C>&& column,
+                                        Mixed&& value)
+{
+    switch (value.get_type()) {
+        case realm::type_Int:
+        case realm::type_Timestamp:
+        case realm::type_Double:
+        case realm::type_Float:
+        case realm::type_Decimal:
+            add_numeric_constraint(RLMPropertyTypeAny, operatorType, column, value);
+            break;
+        case realm::type_UUID:
+        case realm::type_ObjectId:
+        case realm::type_Bool:
+            add_bool_constraint(RLMPropertyTypeAny, operatorType, column, value);
+            break;
+        case realm::type_String:
+        case realm::type_Binary:
+            add_string_constraint(operatorType, predicateOptions, std::move(column), value);
+            break;
+        default:
+            @throw RLMException(@"Unsupported RLMValue type.");
+    }
+}
```

Comment:
Similarly to above, I don't think the value that a Mixed happens to have should have an effect on what operations are legal on it.

Reply:
We need to make sense of what the comparing value is in each case and what constraint should be assigned. For example if I pass in `@123.456` how would we differentiate that from `@"hello"`?